### DISCUSSION TITLE
Error on DownloadFileAsync with sandbox files id

### DISCUSSION THREAD

**Discussion by sliberali1986** [RageAgainstThePixel/OpenAI-DotNet#379] [Q&A] [answered]
# Bug Report

## Overview
Unable to download sanbox files with FilesEndpoint.DownloadFileAsync.

## To Reproduce

1. Ask AI to create a file (for example a pptx file).
2. The MessageResponse object contains a message like "Download file here: (sandbox:/mnt/data/total_amount_by_year.pptx)" and the Attachments property contains the File Id.
3.  The file response obtained with FilesEndpoint.GetFileInfoAsync is valid. The FileName property has the value "/mnt/data/total_amount_by_year.pptx".
4. Call the FilesEndpoint.DownloadFileAsync to save binary.
5. Exception: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\mnt\data\total_amount_by_year.pptx'.

## Expected behavior
Everything works and the binary will be saved in file system.

## Screenshots
![Image](https://github.com/user-attachments/assets/6e00a5d4-9b5c-43b8-9b2a-6d1a3f4eac0b)

**Comment by StephenHodgson** [comment]
Likely you need to use RetrieveFileStreamAsync option so you can save the contents to disk yourself.

Is this in a browser app, console, or embedded somewhere else?

**Comment by sliberali1986** [comment, marked as answer]
Hi @StephenHodgson,
I'm using the package in a WebAPI. The directory is local (with read/write permission). The problem is the Path.Combine when FileResponse.FileName contains something like "/mnt/data/....". Check my screenshot:

![Image](https://github.com/user-attachments/assets/47804c5d-0821-4d94-815d-8e61522309e7)

The Path.Combine returns a value that ignore the passed directory and after it try to write the file under C:\mnt\data... with exception. I solved the issue using the RetrieveFileStreamAsync and wrapping the FileResponse.FileName with Path.GetFileName to obtain a valid file path. Thanks for your help.
